packages/payload/src/collections/graphql/init.ts

```diff
@@ -76,7 +77,10 @@ function initCollectionsGraphQL(payload: Payload): void {
 
     collection.graphQL = {} as Collection['graphQL']
 
-    const idField = fields.find((field) => fieldAffectsData(field) && field.name === 'id')
+    const idField =
```

Comment:
This variable is not named appropriately. You are setting this const to a boolean, but on reading the code, I was thinking that it was the ID field itself. I know this is a problem with the code before you came in, but just some quick feedback for the future - feel free to rename any weird variables like this to `hasIDField` or similar if you find them.